### DISCUSSION TITLE
Parsing workspace specifications without POIs

### DISCUSSION THREAD

**Discussion by alexander-held** [scikit-hep/pyhf#1602] [Q&A] [answered]
https://github.com/scikit-hep/pyhf/pull/950 introduced the possibility to build models without POIs. There are two ways to use this feature that I am aware of:
- start with a workspace containing a POI, and build a model from it with `poi_name=None`,
- build the model directly, bypassing validation of the measurement config schema.

It is however not possible to start with a workspace specification that does not include the POI, as the workspace schema requires it. Are there some useful workarounds for this limitation? Workspaces without POIs can for example arise dynamically due to pruning of samples (e.g. the signal sample, cc @Daniel-Noel). Pruned workspaces can then be difficult to use.

Would it potentially make sense to allow parsing a workspace while bypassing the schema validation to help with this use case? Instead of bypassing the schema completely, maybe a special handling for the POI could be added when `poi_name=None` is used and no POI is present?

An example to demonstrate things in more detail, using `pyhf` 0.6.3:
```python
import copy
import pyhf

spec = {
    "channels": [
        {
            "name": "SR",
            "samples": [
                {
                    "data": [51.8],
                    "modifiers": [
                        {"data": [2.6], "name": "staterror_SR", "type": "staterror"},
                        {"data": None, "name": "mu", "type": "normfactor"},
                    ],
                    "name": "signal",
                }
            ],
        }
    ],
    "measurements": [
        {
            "config": {"parameters": [], "poi": "mu"},
            "name": "example",
        }
    ],
    "observations": [{"data": [52], "name": "SR"}],
    "version": "1.0.0",
}

model = pyhf.Workspace(spec).model()  # works fine
model = pyhf.Workspace(spec).model(poi_name=None)  # works fine

spec_without_poi = copy.deepcopy(spec)
spec_without_poi["measurements"][0]["config"].pop("poi")
# model = pyhf.Workspace(spec_without_poi).model()  # does not work
# model = pyhf.Workspace(spec_without_poi).model(poi_name=None)  # does not work either

# workaround: extract model specification
model_spec = {"channels": spec_without_poi["channels"]}
model = pyhf.Model(model_spec, poi_name=None)
```

Construction of a model from the original specification works fine. After removal of the POI, the workspace corresponding to the new `spec_without_poi` can no longer be constructed, as the workspace schema is now violated (first two lines that are commented out).

It is possible to build a model schema and then build the model directly from that, however this approach has a big downside: other measurement config options (for example inits of parameters, or setting parameters to constant, or parameter ranges) cannot be propagated to a model created in this manner. Is there a workaround for this limitation?

**Comment by kratsg** [comment, marked as answer]
See #1638.

**Reply by alexander-held** [reply]
With that change, the following works:
```python
spec_without_poi["measurements"][0]["config"]["poi"] = ""
model = pyhf.Workspace(spec_without_poi).model()
model = pyhf.Workspace(spec_without_poi).model(poi_name=None)  # does the same
```
